### PR TITLE
Add .NET 6+ template for Android wear

### LINKED ISSUE #7003: WearOS app fails to compile when targeting net6.0-android
### Android application type

Android for .NET (net6.0-android, etc.)

### Affected platform version

VS4Mac 2022 Preview (17.0 build 8904), .NET 6.0.300, Xamarin.Android 12.3.0.3

### Description

I have been trying to use net6.0-android as the target framework to build a WearOS wearable app after it was mentioned in an (unofficial) [.NET MAUI Platforms Support gist](https://gist.github.com/vhugogarcia/4cacaa90b42495c21a62044efc298beb)](https://gist.github.com/vhugogarcia/4cacaa90b42495c21a62044efc298beb) that it was supported.

The warnings below seem to indicate `android-31/optional/com.google.android.wearable.jar` is missing when in actual fact `com.google.android.wearable.jar` does not exist in `android-30/optional/` and `android-28/optional/` which are the other folders I have there.

I am unsure if those warnings relate to the R8 compilation or not.

Repro: 
[WearOSMonoTest.zip](https://github.com/xamarin/xamarin-android/files/8699240/WearOSMonoTest.zip)
[WearOSNet6Test.zip](https://github.com/xamarin/xamarin-android/files/8699248/WearOSNet6Test.zip)

Extra info:
I did update target framework on the first project to MonoAndroid12.0
I also updated `TargetFramework` to `net6.0-android31` because that is what was listed as the required framework on nuget.org for the nugets I added but I don't believe this is required as it has the same issues if I use net6.0-android.

### Steps to Reproduce

1. Create new wear app using VS4Mac stable (in this case I called it WearOSMonoTest)
2. Update all nuget files.
3. Build and deploy app to confirm it is working.
4. (see WearOSMonoTest.zip above)
5. Create new net6.0-android app, `dotnet new android --name WearOSNet6Test`
6. Update nugets/resources/code to match the `WearOSMonoTest` project. 
7. Build and deploy app to see .NET 6 in a WearOS app.
8. Get compile issues (listed below)
9. (see WearOSNet6Test.zip above)

### Did you find any workaround?

Nope.

### Relevant log output

```shell
obj/Debug/net6.0-android31/android/AndroidManifest.xml : warning XA4218: Unable to find //manifest/application/uses-library at path: /Users/bradmoore/Library/Developer/Xamarin/android-sdk-macosx/platforms/android-31/optional/com.google.android.wearable.jar
obj/Debug/net6.0-android31/android/AndroidManifest.xml : warning XA4218: Unable to find //manifest/application/uses-library at path: /Users/bradmoore/Library/Developer/Xamarin/android-sdk-macosx/platforms/android-31/optional/androidx.window.extensions.jar
obj/Debug/net6.0-android31/android/AndroidManifest.xml : warning XA4218: Unable to find //manifest/application/uses-library at path: /Users/bradmoore/Library/Developer/Xamarin/android-sdk-macosx/platforms/android-31/optional/androidx.window.sidecar.jar
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/32.0.300-rc.3.45/tools/Xamarin.Android.D8.targets(79,5): error : java.lang.RuntimeException: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: obj/Debug/net6.0-android31/lp/59/jl/classes.jar : android/support/v4/app/INotificationSideChannel$Stub.class
```
```

### PR DESCRIPTION
**Pull request by dellis1972**
Fixes #7003

This speaks for itself, lets add a template for Wear applications.

### CODE REVIEW

**Review by jonathanpeppers — COMMENTED** [review]
Can we add this new template to this test?

https://github.com/xamarin/xamarin-android/blob/b250c04b09a2b725336ae6d6c5693e0b9f37e4cc/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/XASdkTests.cs#L214-L215

File: src/Microsoft.Android.Templates/android-wear/AndroidApp1.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <ApplicationId>com.companyname.AndroidApp1</ApplicationId>
+    <ApplicationVersion>1</ApplicationVersion>
+    <ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Xamarin.AndroidX.Wear" Version="1.2.0.5" />
+    <PackageReference Include="Xamarin.AndroidX.PercentLayout" Version="1.0.0.14" />
+    <PackageReference Include="Xamarin.AndroidX.Legacy.Support.Core.UI" Version="1.0.0.14" />
```
[the remark this answers is not in the document]
Comment (dellis1972) [reply src/Microsoft.Android.Templates/android-wear/AndroidApp1.csproj:16]: I'll test with just the Wear package. This is based on the working project the customer provided.

File: src/Microsoft.Android.Templates/android-wear/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1" android:versionName="1.0">
+	<uses-feature android:name="android.hardware.type.watch" />
+	<uses-permission android:name="android.permission.WAKE_LOCK" />
+	<application android:allowBackup="true" android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:supportsRtl="true" android:theme="@android:style/Theme.DeviceDefault">
+		<uses-library android:name="com.google.android.wearable" android:required="true" />
+		<!--
+		Set to true if your app is Standalone, that is, it does not require the handheld
+		app to run.
+		-->
+		<meta-data android:name="com.google.android.wearable.standalone" android:value="true" />
```
[the remark this answers is not in the document]
Comment (dellis1972) [reply src/Microsoft.Android.Templates/android-wear/AndroidManifest.xml:11]: again, this is based on the customer template. I'll remove this.

File: src/Microsoft.Android.Templates/android-wear/Resources/values/strings.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="app_name">AndroidApp1</string>
+    <!--
+    This string is used for square devices and overridden by hello_world in
+    values-round/strings.xml for round devices.
+    -->
+    <string name="app_text">Hello, Android!</string>
```
Comment (jonathanpeppers) [inline src/Microsoft.Android.Templates/android-wear/Resources/values/strings.xml:8]: Should it say:
```suggestion
    <string name="app_text">Hello, Android Wear!</string>
```

**Review by jonathanpeppers — APPROVED** [review]
Otherwise LGTM 👍

**Comment by jonpryor** [comment]
The fundamental question is, do we actually want to provide a template for Wear *at all*?

Providing a template implies that it's "fit for purpose", is tested, is maintained, will continue to work.

Which, for starters, implies that we'd have a "WearOS 'device'" that we use as a target, as we do with Android emulators now.

Do we want to actually do that?

VSMac *used* to have a Wear template.  VSMac 17 *does not* have a Wear template.  I suspect VSWin should go the same way (remove the template), and I'm not entirely certain that *we* want to provide a template *either*.

**Comment by jonathanpeppers** [comment]
A test fails:
```
  Failed DotNetNew("androidwear") [1 m 1 s]
  Error Message:
     /Users/runner/work/1/a/TestRelease/05-19_15.06.09/temp/DotNetNewandroidwear/build.log should have no MSBuild warnings.
```

**Comment by dellis1972** [comment]
> A test fails:
> 
> ```
>   Failed DotNetNew("androidwear") [1 m 1 s]
>   Error Message:
>      /Users/runner/work/1/a/TestRelease/05-19_15.06.09/temp/DotNetNewandroidwear/build.log should have no MSBuild warnings.
> ```

Had to ignore these warnings in the test. It the `XA4218` which is where we look up the `uses-library` stuff from an `AndroidManifest` in a library and check the sdk directory for them. In this case these files do not exists so we end up with this `XA4218` warning.

**Comment by dellis1972** [comment]
@jonpryor is it worth me adding a test step to  install and run the WearOS emulator and run a test on it? It works locally for me, so we could actually start testing at least the template on  the emulator during CI?

**Comment by jonpryor** [comment]
@dellis1972 [asked](https://github.com/xamarin/xamarin-android/pull/7013#issuecomment-1134544212):

> is it worth me adding a test step to install and run the WearOS emulator and run a test on it?

*Yes*.  Yes it is.  If we provide this template at all ([see also](https://github.com/xamarin/xamarin-android/pull/7013#issuecomment-1131952593)), then we're implicitly/explicitly stating that the template is *supported* and *works*.  Consequently, we *should* ensure that this runs as expected on a WearOS emulator for testing.

**Review by jonathanpeppers — CHANGES_REQUESTED** [review]
If we get to a point we want to merge this, we should use the updated .NET-branded app icons here: https://github.com/xamarin/xamarin-android/pull/7065

**Comment by dellis1972** [comment]
> If we get to a point we want to merge this, we should use the updated .NET-branded app icons here: #7065

Icons have been updated 👍

File: src/Microsoft.Android.Templates/android-wear/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1" android:versionName="1.0">
+	<uses-feature android:name="android.hardware.type.watch" />
+	<uses-permission android:name="android.permission.WAKE_LOCK" />
+	<application android:allowBackup="true" android:icon="@mipmap/appicon" android:label="@string/app_name" android:supportsRtl="true" android:theme="@android:style/Theme.DeviceDefault">
```
Comment (jonathanpeppers) [inline src/Microsoft.Android.Templates/android-wear/AndroidManifest.xml:5]: Does this also need `android:roundIcon="@mipmap/appicon_round"`?

Like we have here:

https://github.com/xamarin/xamarin-android/blob/6b43af37bc57b7eea2a213ee15de037e61784a1c/src/Microsoft.Android.Templates/android/AndroidManifest.xml#L3

Would it also be worth sharing the icons in a single folder? And we can make the build pull them in for both templates?

Comment (jonathanpeppers) [reply src/Microsoft.Android.Templates/android-wear/AndroidManifest.xml:5]: I think it uses the same purple square `.png`, but a different `.xml`? Android then clips it to a circle.

**Pull request merged**